runtime/compiler/optimizer/EscapeAnalysis.cpp

```diff
@@ -133,7 +133,7 @@ TR_EscapeAnalysis::TR_EscapeAnalysis(TR::OptimizationManager *manager)
    _dememoizationSymRef = NULL;
 
    _createStackAllocations   = true;
-   _createLocalObjects       = comp()->target().cpu.isX86() || comp()->target().cpu.isPower() || comp()->target().cpu.isZ();
+   _createLocalObjects       = comp()->target().cpu.isX86() || comp()->target().cpu.isPower() || comp()->target().cpu.isZ() || comp()->target().cpu.isARM64();
```

Comment:
Can this growing list of architecture checks (here and elsewhere in this PR) be replaced with a capability query instead?  What are they really asking:  whether a code generator is capable of mapping local objects to the stack?  If so, I think it would be much cleaner to ask that question of a code generator and have each architecture answer true or false.

@vijaysun-omr or @andrewcraik any opinion or insight on whether that's what these architecture checks are really for?

Reply:
Yes I agree with you, though the functionality being checked for is probably the entirety of supporting stack allocations, e.g. mapping local objects to the stack, emitting a stack alloc map etc. I don't see any reason why this cannot be a codegen query.

Reply:
Thanks.

@Akira1Saitoh , can you introduce a `supportsStackAllocations()` query in the common code generator and have each architecture specialize it as necessary (basically, return true for every architecture other than ARM)?

You could probably do this in two commits: one to create the API, and the second to actually use it in place of the architecture checks.

Reply:
Sure, I will add the API to the common code generator and replace those architecture checks with the query.